src/utilities/memory.c

```diff
@@ -62,12 +62,16 @@ hypre_DeviceMemset(void *ptr, HYPRE_Int value, size_t num)
 {
 #if defined(HYPRE_USING_DEVICE_OPENMP)
 #if defined(HYPRE_DEVICE_OPENMP_ALLOC)
-   HYPRE_CUDA_CALL( cudaMemset(ptr, value, num) );
+   #pragma omp target teams distribute parallel for is_device_ptr(ptr)
+   for (size_t i = 0; i < num; i++)
+   {
+      ((unsigned char *) ptr)[i] = (unsigned char) value;
+   }
 #else
    memset(ptr, value, num);
    HYPRE_OMPOffload(hypre__offload_device_num, ptr, num, "update", "to");
 #endif
-   HYPRE_CUDA_CALL( cudaDeviceSynchronize() );
+   /* HYPRE_CUDA_CALL( cudaDeviceSynchronize() ); */
```

Comment:
Here also a lot of calls are commented out. Leaving empty if statements in some cases. Are they no longer needed?